cmake/alpakaCommon.cmake

```diff
@@ -261,278 +261,146 @@ endif()
 #-------------------------------------------------------------------------------
 # Find CUDA.
 if(ALPAKA_ACC_GPU_CUDA_ENABLE)
+    include(CheckLanguage)
+    check_language(CUDA)
 
-    if(NOT DEFINED ALPAKA_CUDA_VERSION)
-        set(ALPAKA_CUDA_VERSION 9.0)
-    endif()
-
-    if(ALPAKA_CUDA_VERSION VERSION_LESS 9.0)
-        message(FATAL_ERROR "CUDA Toolkit < 9.0 is not supported!")
-
-    else()
-        find_package(CUDA "${ALPAKA_CUDA_VERSION}")
-        if(NOT CUDA_FOUND)
-            message(FATAL_ERROR "Optional alpaka dependency CUDA could not be found!")
-        else()
-            set(ALPAKA_CUDA_VERSION "${CUDA_VERSION}")
-            if(CUDA_VERSION VERSION_LESS 10.3)
-                set(ALPAKA_CUDA_ARCH "30" CACHE STRING "GPU architecture")
-            else()
-                set(ALPAKA_CUDA_ARCH "35" CACHE STRING "GPU architecture")
-            endif()
-            set(ALPAKA_CUDA_COMPILER "nvcc" CACHE STRING "CUDA compiler")
-            set_property(CACHE ALPAKA_CUDA_COMPILER PROPERTY STRINGS "nvcc;clang")
+    if(CMAKE_CUDA_COMPILER)
+        enable_language(CUDA)
+        find_package(CUDAToolkit ${ALPAKA_CUDA_VERSION} REQUIRED)
 
-            option(ALPAKA_CUDA_FAST_MATH "Enable fast-math" ON)
-            option(ALPAKA_CUDA_FTZ "Set flush to zero for GPU" OFF)
-            option(ALPAKA_CUDA_SHOW_REGISTER "Show kernel registers and create PTX" OFF)
-            option(ALPAKA_CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps (folder: nvcc_tmp)" OFF)
-            option(ALPAKA_CUDA_NVCC_EXPT_EXTENDED_LAMBDA "Enable experimental, extended host-device lambdas in NVCC" ON)
-            option(ALPAKA_CUDA_NVCC_SEPARABLE_COMPILATION "Enable separable compilation in NVCC" OFF)
+        target_compile_features(alpaka INTERFACE cuda_std_${ALPAKA_CXX_STANDARD})
```

Comment:
I still find it weird that we do not pass `CMAKE_CXX_STANDARD` here. But this can be a topic for a different discussion.

Reply:
With the new native CMake support, `ALPAKA_CXX_STANDARD` is not necessary anymore. alpaka would define its minimum requirements as `cxx_std_14` and `cuda_std_14`. The user is then able to overwrite this with `CMAKE_CXX_STANDARD=17` and `CMAKE_CUDA_STANDARD=17`. This is not as easy as before but the natively supported variant. I am not sure what happens when the user specifies two different versions but my expectation would be that CMake also forwards those different flags to nvcc/gcc respectively.
However, we still have to support HIP. There we explicitly have to pass the `std` flag ourselves. We would have to decide on a version there. We could pass the value of `CMAKE_CXX_STANDARD` to HIP if it is set and else use 14 as default.

However, I would like to do such a big change later independently.

Reply:
Makes perfect sense to delay such a change! Thank you for the explanation!